docs/modules/ROOT/pages/explanations/decisions/api-design.adoc

```diff
@@ -0,0 +1,261 @@
+= Decision on API Design
+
+== Problem
+
+This decision is about how we can achieve a common look as defined https://kb.vshn.ch/app-catalog/explanations/why_exists.html#_standard_way_to_provide_services[here].
+
+The ideal way would be, that we can re-use the exact same XRD for each set of services.
+For example all implementations for PostgreSQL have exactly the same API for the user.
+Due to the nature of the product, not all implementations provide the same functionality.
+The set of available features is most likely not the same between most providers.
+Also with Crossplane we're a bit limited on what's possible with translating the source API spec into the target API spec:
+
+* No conditionals in compositions
+* Only a limited set of transformations
+* Can't create random values
+
+[NOTE]
+This is not a complete list of limitations.
+
+== Requirements
+
+* APIs for a set of similar services feel similar
+* Only minimal configuration is needed for a default case
+* Unique features of a service provider can be leveraged
+
+== Proposals
+
+=== Proposal 1: Use same XRD with provider specific fields
+
+There's exactly one XRD for a set of services.
+This XRD contains a minimal set of common parameters, as well as a field for provider specific configurations.
+
+.Example Claim
+[source, yaml]
+----
+apiVersion: appcat.vshn.io/v1
+kind: PostgreSQL
+metadata:
+  name: my-instance
+spec:
+  parameters:
+    name: my-instance
+    majorVersion: '14'
+    pgSettings: {}
+    exoscale:
+      plan: hobbyist-2
+      type: pg
+      maintenance:
+        day: 2
+        hour: 22
+        minute: 0
+      backup:
+        hour: 20
+        minute: 0
+    vshn:
+      minorVersion: '4'
+----
+
+Advantages::
+
+* We can use the same XRD
+* Adding new providers to the list is easy
+
+Disadvantages::
+
+* We don't really have a "common look"
+* The customer needs to know what the underlying service provider is, even for simple instances
```

Comment:
What's the purpose of mentioning "simple instance"? single-replica? how is that different to a "complex instance"?
```suggestion
* The user needs to know some specifics of the underlying service provider.
```